Linux session crash after Zed launch - GPU hung

### Summary

<!-- Please insert a one line summary of the issue below -->
Since two or three weeks Zed crash at launch on Pop!_OS. At first it was freezing my whole computer, but when I tried yesterday it now crash my session.

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Launch Zed
2. Do some action like opening some files
3. Zed don't act the action I do
4. Screens came black
5. Screens came back but frozen
6. Session crash

Actual Behavior:

Expected Behavior:


### Zed Version and System Specs

```
❯ zed --version
ldconfig: /app/lib/libbsd.so.0 is not a symbolic link

Zed 0.176.2 efe4418b9a2f9279604b6e03a42f204363b0d16e  – /home/user/.local/share/flatpak/app/dev.zed.Zed/x86_64/stable/d651459cf11b98d69c36d8928797d0745bcd3f550825cc913bd14757398a5d3c/files/libexec/zed-editor

❯ screenfetch
                             OS: Pop 22.04 jammy
                             Kernel: x86_64 Linux 6.9.3-76060903-generic
         #####               Uptime: 44m
        #######              Packages: Unknown
        ##O#O##              Shell: zsh 5.8.1
        #######              Resolution: 5120x1440
      ###########            DE: GNOME 41.7
     #############           WM: Mutter
    ###############          WM Theme: Pop
    ################         GTK Theme: Pop-dark [GTK2/3]
   #################         Icon Theme: Pop
 #####################       Font: Fira Sans Semi-Light 10
 #####################       Disk: 237G / 912G (28%)
   #################         CPU: Intel Core i7-6700 @ 8x 4GHz [46.0°C]
                             GPU: AMD/ATI Navi 32 [Radeon RX 7700 XT / 7800 XT]
                             RAM: 10215MiB / 15951MiB
```

Zed is currently installed with **Flatpak**

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-03-04T20:03:16.87636264+01:00 [INFO] ========== starting zed ==========
2025-03-04T20:03:17.177462633+01:00 [INFO] Compositor detection: _NET_WM_CM_S?=false, _NET_WM_CM_OWNER=false, _NET_SUPPORTING_WM_CHECK=true
2025-03-04T20:03:17.177578418+01:00 [INFO] x11: compositor present: true, gtk_frame_extents_supported: true
2025-03-04T20:03:17.322727623+01:00 [INFO] Presentation support: "VK_KHR_wayland_surface"
2025-03-04T20:03:17.323182878+01:00 [INFO] Presentation support: "VK_KHR_xcb_surface"
2025-03-04T20:03:17.323306783+01:00 [INFO] Presentation support: "VK_KHR_xlib_surface"
2025-03-04T20:03:17.323397697+01:00 [INFO] Enabling Vulkan Portability
2025-03-04T20:03:17.323519831+01:00 [INFO] Enabling color space support
2025-03-04T20:03:17.347376935+01:00 [INFO] Adapter: "AMD Radeon RX 7700 XT (RADV NAVI32)"
2025-03-04T20:03:17.347662268+01:00 [INFO] Ray tracing is supported
2025-03-04T20:03:17.365020836+01:00 [INFO] Try connect ibus
2025-03-04T20:03:17.370148458+01:00 [INFO] perform;
2025-03-04T20:03:17.370200712+01:00 [INFO] authenticate;
2025-03-04T20:03:17.370293655+01:00 [INFO] read_command;
2025-03-04T20:03:17.370851356+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:17.372588049+01:00 [INFO] socket reader;
2025-03-04T20:03:17.372666778+01:00 [INFO] read_socket;
2025-03-04T20:03:17.372748406+01:00 [INFO] read_socket;
2025-03-04T20:03:17.372854207+01:00 [INFO] new;
2025-03-04T20:03:17.373080203+01:00 [INFO] read_socket;
2025-03-04T20:03:17.373236014+01:00 [INFO] read_socket;
2025-03-04T20:03:17.373367767+01:00 [INFO] read_socket;
2025-03-04T20:03:17.373777916+01:00 [INFO] read_socket;
2025-03-04T20:03:17.373857608+01:00 [INFO] keep_updated;
2025-03-04T20:03:17.376218379+01:00 [INFO] read_socket;
2025-03-04T20:03:17.377402374+01:00 [INFO] read_socket;
2025-03-04T20:03:17.378426814+01:00 [INFO] read_socket;
2025-03-04T20:03:17.378591064+01:00 [INFO] read_socket;
2025-03-04T20:03:17.378673841+01:00 [INFO] read_socket;
2025-03-04T20:03:17.37880443+01:00 [INFO] read_socket;
2025-03-04T20:03:17.378893937+01:00 [INFO] read_socket;
2025-03-04T20:03:17.379015381+01:00 [INFO] read_socket;
2025-03-04T20:03:17.379112523+01:00 [INFO] read_socket;
2025-03-04T20:03:17.381900397+01:00 [INFO] Opening main db
2025-03-04T20:03:17.392581024+01:00 [INFO] Opening main db
2025-03-04T20:03:17.397414824+01:00 [INFO] Using git binary path: None
2025-03-04T20:03:17.484763204+01:00 [INFO] No prompt template overrides directory found at /home/user/.var/app/dev.zed.Zed/data/zed/prompt_overrides. Using built-in prompts.
2025-03-04T20:03:17.504410552+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2025-03-04T20:03:17.524906244+01:00 [INFO] set environment variables from shell:/bin/zsh, path:/home/user/.asdf/shims:/home/user/.asdf/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
2025-03-04T20:03:17.534040157+01:00 [INFO] activate is not implemented on Linux, ignoring the call
2025-03-04T20:03:17.537645954+01:00 [INFO] Opening main db
2025-03-04T20:03:17.538385932+01:00 [INFO] perform;
2025-03-04T20:03:17.538445909+01:00 [INFO] authenticate;
2025-03-04T20:03:17.538512873+01:00 [INFO] read_command;
2025-03-04T20:03:17.538587815+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:17.538965265+01:00 [INFO] socket reader;
2025-03-04T20:03:17.539025977+01:00 [INFO] read_socket;
2025-03-04T20:03:17.539098806+01:00 [INFO] read_socket;
2025-03-04T20:03:17.545543582+01:00 [INFO] read_socket;
2025-03-04T20:03:17.547221573+01:00 [INFO] read_socket;
2025-03-04T20:03:17.547889645+01:00 [INFO] read_socket;
2025-03-04T20:03:17.548560857+01:00 [INFO] read_socket;
2025-03-04T20:03:17.560233765+01:00 [INFO] Using Visual { id: 111, colormap: 0, depth: 32 }
2025-03-04T20:03:17.560319735+01:00 [INFO] Creating colormap 62914563
2025-03-04T20:03:17.56091918+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2025-03-04T20:03:17.562259042+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:17.565284171+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:03:17.565407082+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:03:17.565484191+01:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2025-03-04T20:03:17.6124017+01:00 [INFO] Refreshing at 6058 micros
2025-03-04T20:03:17.636145158+01:00 [INFO] Using GPU: GpuSpecs { is_software_emulated: false, device_name: "AMD Radeon RX 7700 XT (RADV NAVI32)", driver_name: "radv", driver_info: "Mesa 24.0.3-1pop1~1711635559~22.04~7a9f319" }
2025-03-04T20:03:17.677620357+01:00 [INFO] XConnected server on 29360135, transport version: 0.0, TRANSPORT_MAX: 20
2025-03-04T20:03:17.68346111+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:17.683536472+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-04T20:03:17.691003712+01:00 [INFO] Opening main db
2025-03-04T20:03:17.704839582+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:17.704901191+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-04T20:03:17.721662525+01:00 [INFO] Opening main db
2025-03-04T20:03:17.729151854+01:00 [INFO] open paths ["/home/user/Projects/homelab/ansible-collection/roles/adguardhome/templates/adguardhome.yml.j2"]
2025-03-04T20:03:17.767601862+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.768803831+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.769545392+01:00 [INFO] attempting to start language server "yaml-language-server", path: "/home/user/Projects/homelab/ansible-playbooks", id: 1
2025-03-04T20:03:17.836887938+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.856709245+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.858689534+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.860321531+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.871588258+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:17.954373801+01:00 [INFO] using project environment variables shell launched in "/home/user/Projects/homelab/ansible-playbooks". PATH="/home/user/.asdf/shims:/home/user/.asdf/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-04T20:03:17.979276856+01:00 [INFO] fetching latest version of language server "yaml-language-server"
2025-03-04T20:03:18.15870542+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:18.218972696+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:18.256629002+01:00 [INFO] Using Visual { id: 111, colormap: 0, depth: 32 }
2025-03-04T20:03:18.256713426+01:00 [INFO] Creating colormap 62914629
2025-03-04T20:03:18.259253256+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:18.261779876+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2025-03-04T20:03:18.263302931+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:03:18.263451877+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:03:18.263522589+01:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2025-03-04T20:03:18.302912839+01:00 [INFO] Refreshing at 6058 micros
2025-03-04T20:03:18.306463283+01:00 [INFO] Using GPU: GpuSpecs { is_software_emulated: false, device_name: "AMD Radeon RX 7700 XT (RADV NAVI32)", driver_name: "radv", driver_info: "Mesa 24.0.3-1pop1~1711635559~22.04~7a9f319" }
2025-03-04T20:03:18.307295891+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:18.389396448+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:18.389459244+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-04T20:03:18.43607729+01:00 [INFO] Using surface present mode MAILBOX
2025-03-04T20:03:18.436151497+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-04T20:03:18.446400319+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:18.670645863+01:00 [INFO] ignoring macOS specific set_edited
2025-03-04T20:03:18.672062818+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/node", working directory: "/", args: ["/home/user/.var/app/dev.zed.Zed/data/zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2025-03-04T20:03:18.696406927+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:18.877012446+01:00 [INFO] using project environment variables shell launched in "/home/user/Projects/zasso/typo". PATH="/home/user/.asdf/shims:/home/user/.asdf/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-04T20:03:19.036567257+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:20.200705287+01:00 [ERROR] theme not found: Rosé Pine
2025-03-04T20:03:20.201174465+01:00 [INFO] attempting to start language server "docker-compose", path: "/home/user/Projects/homelab/ansible-playbooks", id: 2
2025-03-04T20:03:20.201328847+01:00 [INFO] attempting to start language server "terraform-ls", path: "/home/user/Projects/homelab/ansible-playbooks", id: 3
2025-03-04T20:03:20.201464764+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:20.20284155+01:00 [INFO] attempting to start language server "rubocop", path: "/home/user/Projects/zasso/typo", id: 4
2025-03-04T20:03:20.202936436+01:00 [INFO] attempting to start language server "ruby-lsp", path: "/home/user/Projects/zasso/typo", id: 5
2025-03-04T20:03:20.256585211+01:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2025-03-04T20:03:20.257252206+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/bundle", working directory: "/home/user/Projects/zasso/typo", args: ["exec", "rubocop", "--lsp"]
2025-03-04T20:03:20.259084839+01:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2025-03-04T20:03:20.259246235+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/ruby-lsp", working directory: "/home/user/Projects/zasso/typo", args: []
2025-03-04T20:03:20.546985291+01:00 [INFO] starting language server process. binary path: "/home/user/.var/app/dev.zed.Zed/data/zed/extensions/work/terraform/terraform-ls-v0.36.4/terraform-ls", working directory: "/home/user/Projects/homelab/ansible-playbooks", args: ["serve"]
2025-03-04T20:03:21.355910064+01:00 [INFO] language server "yaml-language-server" is already installed
2025-03-04T20:03:21.356177923+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/node", working directory: "/home/user/Projects/homelab/ansible-playbooks", args: ["/home/user/.var/app/dev.zed.Zed/data/zed/languages/yaml-language-server/node_modules/yaml-language-server/bin/yaml-language-server", "--stdio"]
2025-03-04T20:03:22.709173249+01:00 [INFO] perform;
2025-03-04T20:03:22.709240708+01:00 [INFO] authenticate;
2025-03-04T20:03:22.709307226+01:00 [INFO] read_command;
2025-03-04T20:03:22.709517718+01:00 [INFO] perform;
2025-03-04T20:03:22.709571568+01:00 [INFO] authenticate;
2025-03-04T20:03:22.709626269+01:00 [INFO] read_command;
2025-03-04T20:03:22.70972305+01:00 [INFO] perform;
2025-03-04T20:03:22.709772317+01:00 [INFO] authenticate;
2025-03-04T20:03:22.709833988+01:00 [INFO] read_command;
2025-03-04T20:03:22.70989035+01:00 [INFO] perform;
2025-03-04T20:03:22.709993903+01:00 [INFO] authenticate;
2025-03-04T20:03:22.710068785+01:00 [INFO] read_command;
2025-03-04T20:03:22.711569654+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.711736717+01:00 [INFO] perform;
2025-03-04T20:03:22.711793732+01:00 [INFO] authenticate;
2025-03-04T20:03:22.711861729+01:00 [INFO] read_command;
2025-03-04T20:03:22.712245576+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.71242055+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.712532734+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.712734668+01:00 [INFO] socket reader;
2025-03-04T20:03:22.712791164+01:00 [INFO] read_socket;
2025-03-04T20:03:22.713459062+01:00 [INFO] perform;
2025-03-04T20:03:22.713519955+01:00 [INFO] authenticate;
2025-03-04T20:03:22.713585885+01:00 [INFO] read_command;
2025-03-04T20:03:22.714094071+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.714162528+01:00 [INFO] socket reader;
2025-03-04T20:03:22.714215967+01:00 [INFO] read_socket;
2025-03-04T20:03:22.714306505+01:00 [INFO] read_socket;
2025-03-04T20:03:22.715067568+01:00 [INFO] read_socket;
2025-03-04T20:03:22.7171634+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:22.717770785+01:00 [INFO] socket reader;
2025-03-04T20:03:22.717826674+01:00 [INFO] read_socket;
2025-03-04T20:03:22.717900034+01:00 [INFO] read_socket;
2025-03-04T20:03:22.719560035+01:00 [INFO] socket reader;
2025-03-04T20:03:22.719642984+01:00 [INFO] read_socket;
2025-03-04T20:03:22.719722085+01:00 [INFO] read_socket;
2025-03-04T20:03:22.719827009+01:00 [INFO] socket reader;
2025-03-04T20:03:22.71988042+01:00 [INFO] read_socket;
2025-03-04T20:03:22.719960084+01:00 [INFO] read_socket;
2025-03-04T20:03:22.723439569+01:00 [INFO] socket reader;
2025-03-04T20:03:22.723502783+01:00 [INFO] read_socket;
2025-03-04T20:03:22.723575712+01:00 [INFO] read_socket;
2025-03-04T20:03:22.732882845+01:00 [INFO] read_socket;
2025-03-04T20:03:22.741130616+01:00 [INFO] read_socket;
2025-03-04T20:03:22.742085105+01:00 [ERROR] theme not found: Rosé Pine
2025-03-04T20:03:22.742649404+01:00 [ERROR] theme not found: Rosé Pine
2025-03-04T20:03:22.743490544+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:03:22.745473915+01:00 [INFO] read_socket;
2025-03-04T20:03:22.751455419+01:00 [INFO] read_socket;
2025-03-04T20:03:22.757821679+01:00 [INFO] read_socket;
2025-03-04T20:03:22.761504607+01:00 [INFO] read_socket;
2025-03-04T20:03:22.762966622+01:00 [INFO] read_socket;
2025-03-04T20:03:22.763120421+01:00 [INFO] read_socket;
2025-03-04T20:03:22.763221029+01:00 [INFO] read_socket;
2025-03-04T20:03:22.763367885+01:00 [INFO] read_socket;
2025-03-04T20:03:22.763422114+01:00 [INFO] read_socket;
2025-03-04T20:03:22.763484868+01:00 [INFO] read_socket;
2025-03-04T20:03:22.764134024+01:00 [INFO] read_socket;
2025-03-04T20:03:22.764284435+01:00 [INFO] read_socket;
2025-03-04T20:03:22.764609118+01:00 [INFO] read_socket;
2025-03-04T20:03:22.764857992+01:00 [INFO] read_socket;
2025-03-04T20:03:22.765221138+01:00 [INFO] read_socket;
2025-03-04T20:03:22.765332074+01:00 [INFO] read_socket;
2025-03-04T20:03:22.765475956+01:00 [INFO] read_socket;
2025-03-04T20:03:22.766165511+01:00 [INFO] read_socket;
2025-03-04T20:03:22.768344188+01:00 [INFO] read_socket;
2025-03-04T20:03:22.76902693+01:00 [INFO] read_socket;
2025-03-04T20:03:22.769145337+01:00 [INFO] read_socket;
2025-03-04T20:03:22.769678921+01:00 [INFO] read_socket;
2025-03-04T20:03:22.830717032+01:00 [INFO] using project environment variables shell launched in "/home/user/Projects/homelab/ansible-collection/roles/adguardhome/templates/adguardhome.yml.j2". PATH="/home/user/.asdf/shims:/home/user/.asdf/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-04T20:03:23.369352145+01:00 [INFO] perform;
2025-03-04T20:03:23.369419575+01:00 [INFO] authenticate;
2025-03-04T20:03:23.369487155+01:00 [INFO] read_command;
2025-03-04T20:03:23.36968697+01:00 [INFO] perform;
2025-03-04T20:03:23.369738407+01:00 [INFO] authenticate;
2025-03-04T20:03:23.369846475+01:00 [INFO] read_command;
2025-03-04T20:03:23.369968847+01:00 [INFO] perform;
2025-03-04T20:03:23.37002823+01:00 [INFO] authenticate;
2025-03-04T20:03:23.37018705+01:00 [INFO] perform;
2025-03-04T20:03:23.370244404+01:00 [INFO] authenticate;
2025-03-04T20:03:23.3703108+01:00 [INFO] read_command;
2025-03-04T20:03:23.370364134+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.370470244+01:00 [INFO] perform;
2025-03-04T20:03:23.370524404+01:00 [INFO] authenticate;
2025-03-04T20:03:23.370584262+01:00 [INFO] read_command;
2025-03-04T20:03:23.370642101+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.37072153+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.370805267+01:00 [INFO] perform;
2025-03-04T20:03:23.37085839+01:00 [INFO] authenticate;
2025-03-04T20:03:23.370922688+01:00 [INFO] read_command;
2025-03-04T20:03:23.371117508+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.371322773+01:00 [INFO] socket reader;
2025-03-04T20:03:23.371377196+01:00 [INFO] read_socket;
2025-03-04T20:03:23.371440019+01:00 [INFO] read_socket;
2025-03-04T20:03:23.371653543+01:00 [INFO] socket reader;
2025-03-04T20:03:23.371706212+01:00 [INFO] read_socket;
2025-03-04T20:03:23.371767123+01:00 [INFO] read_socket;
2025-03-04T20:03:23.37197927+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.372033072+01:00 [INFO] socket reader;
2025-03-04T20:03:23.372088672+01:00 [INFO] read_socket;
2025-03-04T20:03:23.372154387+01:00 [INFO] read_socket;
2025-03-04T20:03:23.372212596+01:00 [INFO] read_command;
2025-03-04T20:03:23.374598473+01:00 [INFO] send_secondary_commands;
2025-03-04T20:03:23.374889645+01:00 [INFO] socket reader;
2025-03-04T20:03:23.374956242+01:00 [INFO] read_socket;
2025-03-04T20:03:23.375026191+01:00 [INFO] read_socket;
2025-03-04T20:03:23.376536036+01:00 [INFO] socket reader;
2025-03-04T20:03:23.376594802+01:00 [INFO] read_socket;
2025-03-04T20:03:23.376662216+01:00 [INFO] read_socket;
2025-03-04T20:03:23.376763505+01:00 [INFO] socket reader;
2025-03-04T20:03:23.376810438+01:00 [INFO] read_socket;
2025-03-04T20:03:23.376874293+01:00 [INFO] read_socket;
2025-03-04T20:03:23.387613983+01:00 [INFO] read_socket;
2025-03-04T20:03:23.391159309+01:00 [INFO] read_socket;
2025-03-04T20:03:23.395653309+01:00 [INFO] read_socket;
2025-03-04T20:03:23.398856799+01:00 [INFO] read_socket;
2025-03-04T20:03:23.399680582+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/node", working directory: "/home/user/Projects/homelab/ansible-playbooks", args: ["/home/user/.var/app/dev.zed.Zed/data/zed/extensions/work/docker-compose/node_modules/.bin/docker-compose-langserver", "--stdio"]
2025-03-04T20:03:23.403244638+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408393454+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408459521+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408531645+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408602906+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408749334+01:00 [INFO] read_socket;
2025-03-04T20:03:23.408804765+01:00 [INFO] read_socket;
2025-03-04T20:03:23.409369338+01:00 [INFO] read_socket;
2025-03-04T20:03:23.409654778+01:00 [INFO] read_socket;
2025-03-04T20:03:23.409952234+01:00 [INFO] read_socket;
2025-03-04T20:03:23.41013832+01:00 [INFO] read_socket;
2025-03-04T20:03:23.410319674+01:00 [INFO] read_socket;
2025-03-04T20:03:23.410385311+01:00 [INFO] read_socket;
2025-03-04T20:03:23.410526983+01:00 [INFO] read_socket;
2025-03-04T20:03:23.410760222+01:00 [INFO] read_socket;
2025-03-04T20:03:23.411730601+01:00 [INFO] read_socket;
2025-03-04T20:03:23.412286529+01:00 [INFO] read_socket;
2025-03-04T20:03:23.412619707+01:00 [INFO] read_socket;
2025-03-04T20:03:23.412785414+01:00 [INFO] read_socket;
2025-03-04T20:03:23.415351794+01:00 [INFO] read_socket;
2025-03-04T20:18:46.588632055+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:18:46.601664181+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-04T20:18:46.606888558+01:00 [INFO] Creating a descriptor pool for at most 256 sets
2025-03-04T20:18:46.612938569+01:00 [INFO] Creating a descriptor pool for at most 256 sets
2025-03-04T20:18:59.257562441+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:03.859456855+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:04.249557248+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:06.170454027+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:07.840042341+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:07.986784421+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:08.115142197+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:08.265934607+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:08.404152709+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:08.947263717+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:09.915455445+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:12.814577837+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:13.981862941+01:00 [INFO] open paths ["/home/user/.var/app/dev.zed.Zed/config/zed/settings.json"]
2025-03-04T20:19:14.059511932+01:00 [ERROR] theme not found: Rosé Pine
2025-03-04T20:19:14.05978437+01:00 [ERROR] theme not found: Rosé Pine
2025-03-04T20:19:14.060203408+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:14.063762325+01:00 [INFO] attempting to start language server "json-language-server", path: "/home/user/.var/app/dev.zed.Zed/config/zed/settings.json", id: 6
2025-03-04T20:19:14.063828972+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-04T20:19:14.172117767+01:00 [INFO] using project environment variables shell launched in "/home/user/.var/app/dev.zed.Zed/config/zed/settings.json". PATH="/home/user/.asdf/shims:/home/user/.asdf/bin:/home/user/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-04T20:19:14.172295412+01:00 [INFO] fetching latest version of language server "json-language-server"
2025-03-04T20:19:15.660559664+01:00 [INFO] language server "json-language-server" is already installed
2025-03-04T20:19:15.660748973+01:00 [INFO] starting language server process. binary path: "/home/user/.asdf/shims/node", working directory: "/home/user/.var/app/dev.zed.Zed/config/zed", args: ["/home/user/.var/app/dev.zed.Zed/data/zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2025-03-04T20:19:27.991124534+01:00 [ERROR] GPU hung

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

<details><summary>Boot log</summary>

```
févr. 18 11:23:00 pop-os kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx_0.0.0 timeout, signaled seq=1712234, emitted seq=1712236
févr. 18 11:23:00 pop-os kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process zed-editor pid 20885 thread zed-editor pid 20885
févr. 18 11:23:00 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: GPU reset begin!
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:01 pop-os kernel: [drm:mes_v11_0_submit_pkt_and_poll_completion.constprop.0 [amdgpu]] *ERROR* MES failed to response msg=3
févr. 18 11:23:01 pop-os kernel: [drm:amdgpu_mes_unmap_legacy_queue [amdgpu]] *ERROR* failed to unmap legacy queue
févr. 18 11:23:02 pop-os kernel: [drm:gfx_v11_0_cp_gfx_enable.isra.0 [amdgpu]] *ERROR* failed to halt cp gfx
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: MODE1 reset
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: GPU mode1 reset
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: GPU smu mode1 reset
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: GPU reset succeeded, trying to resume
févr. 18 11:23:02 pop-os kernel: [drm] PCIE GART of 512M enabled (table at 0x0000008000300000).
févr. 18 11:23:02 pop-os kernel: [drm] VRAM is lost due to GPU reset!
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: PSP is resuming...
févr. 18 11:23:02 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: reserve 0xa700000 from 0x82e0000000 for PSP TMR
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: RAP: optional rap ta ucode is not available
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: SECUREDISPLAY: securedisplay ta ucode is not available
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: SMU is resuming...
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: smu driver if version = 0x0000003d, smu fw if version = 0x00000040, smu fw program = 0, smu fw version = 0x00504f00 (80.79.0)
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: SMU driver if version not matched
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: SMU is resumed successfully!
févr. 18 11:23:03 pop-os kernel: [drm] DMUB hardware initialized: version=0x07002900
févr. 18 11:23:03 pop-os kernel: [drm] kiq ring mec 3 pipe 1 q 0
févr. 18 11:23:03 pop-os kernel: [drm] VCN decode and encode initialized successfully(under DPG Mode).
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: [drm:jpeg_v4_0_hw_init [amdgpu]] JPEG decode initialized successfully.
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring gfx_0.0.0 uses VM inv eng 0 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 6 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 7 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 8 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 9 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 10 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 11 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring sdma0 uses VM inv eng 12 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring sdma1 uses VM inv eng 13 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring vcn_unified_0 uses VM inv eng 0 on hub 8
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring vcn_unified_1 uses VM inv eng 1 on hub 8
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring jpeg_dec uses VM inv eng 4 on hub 8
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: ring mes_kiq_3.1.0 uses VM inv eng 14 on hub 0
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: recover vram bo from shadow start
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: recover vram bo from shadow done
févr. 18 11:23:03 pop-os kernel: amdgpu 0000:03:00.0: amdgpu: GPU reset(2) succeeded!
févr. 18 11:23:03 pop-os firefox.desktop[2050]: amdgpu: amdgpu_cs_query_fence_status failed.
févr. 18 11:23:03 pop-os firefox.desktop[2050]: [GFX1-]: Detect DeviceReset DeviceResetReason::DRIVER_ERROR DeviceResetDetectPlace::WR_POST_UPDATE in Parent process
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: amdgpu: The CS has been rejected (-125).
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) Backtrace:
févr. 18 11:23:03 pop-os gnome-shell[1342]: amdgpu: amdgpu_cs_query_fence_status failed.
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 0: /usr/lib/xorg/Xorg (OsLookupColor+0x139) [0x62a7db485ad9]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 1: /lib/x86_64-linux-gnu/libc.so.6 (__sigaction+0x50) [0x79af7b242520]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (pthread_kill+0x12c) [0x79af7b2969fc]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 3: /lib/x86_64-linux-gnu/libc.so.6 (raise+0x16) [0x79af7b242476]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 4: /lib/x86_64-linux-gnu/libc.so.6 (abort+0xd3) [0x79af7b2287f3]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 5: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so (amdgpu_winsys_create+0x49f4) [0x79af7911a5d4]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 6: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so (amdgpu_winsys_create+0x7469) [0x79af7911d049]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 7: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so (__driDriverGetExtensions_zink+0x48a58) [0x79af788f21b8]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 8: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so (__driDriverGetExtensions_zink+0x67f9b) [0x79af789116fb]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 9: /lib/x86_64-linux-gnu/libc.so.6 (pthread_condattr_setpshared+0x513) [0x79af7b294ac3]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) 10: /lib/x86_64-linux-gnu/libc.so.6 (__xmknodat+0x230) [0x79af7b326850]
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: Fatal server error:
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) Caught signal 6 (Aborted). Server aborting
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: Please consult the The X.Org Foundation support
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]:          at http://wiki.x.org
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]:  for help.
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) Please also check the log file at "/home/user/.local/share/xorg/Xorg.0.log" for additional information.
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE)
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (II) event5  - Logitech G403 Prodigy Gaming Mouse: SYN_DROPPED event - some input events have been lost.
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (EE) event7  - Logitech PRO Gaming Keyboard: client bug: event processing lagging behind by 1263ms, your system is too slow
févr. 18 11:23:03 pop-os /usr/libexec/gdm-x-session[1174]: (II) AIGLX: Suspending AIGLX clients for VT switch
févr. 18 11:25:01 pop-os CRON[53677]: pam_unix(cron:session): session opened for user root(uid=0) by (uid=0)
févr. 18 11:25:01 pop-os CRON[53678]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
févr. 18 11:25:01 pop-os CRON[53677]: pam_unix(cron:session): session closed for user root
févr. 18 11:25:51 pop-os kernel: sysrq: This sysrq operation is disabled.
févr. 18 11:25:54 pop-os kernel: sysrq: This sysrq operation is disabled.
févr. 18 11:25:56 pop-os kernel: sysrq: This sysrq operation is disabled.
févr. 18 11:25:58 pop-os kernel: sysrq: Emergency Sync
févr. 18 11:25:58 pop-os kernel: Emergency Sync complete
févr. 18 11:26:00 pop-os kernel: sysrq: Emergency Remount R/O
```
</details>

Similar issue. OS freezes, screen goes black, and sometimes returns but is freezed. AMD system

```
zed --version
Zed 0.176.2 efe4418b9a2f9279604b6e03a42f204363b0d16e  – /home/david/.local/zed.app/libexec/zed-editor


                             OS: Pop 22.04 jammy
                             Kernel: x86_64 Linux 6.9.3-76060903-generic
         #####               Uptime: 6h 13m
        #######              Packages: Unknown
        ##O#O##              Shell: bash 5.1.16
        #######              Resolution: 3840x2160
      ###########            DE: GNOME 41.7
     #############           WM: Mutter
    ###############          WM Theme: Pop
    ################         GTK Theme: Pop-dark [GTK2/3]
   #################         Icon Theme: Pop
 #####################       Font: Fira Sans Semi-Light 10
 #####################       Disk: 218G / 3.7T (7%)
   #################         CPU: AMD Ryzen 9 8945HS w/ Radeon 780M Graphics @ 16x 5.263GHz
                             GPU: AMD Radeon Graphics (radeonsi, gfx1103_r1, LLVM 15.0.7, DRM 3.57, 6.9.3-76060903-generic)
                             RAM: 10774MiB / 94320MiB

```

<details>

<summary>Logs</summary>

```
2025-03-05T23:33:38.62206242+01:00 [INFO] ignoring macOS specific set_edited
2025-03-05T23:34:59.933739464+01:00 [ERROR] auto-update failed: error:failed to fetch release: "error code: 524"
2025-03-05T23:36:12.876704232+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:36:12.876876263+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:36:12.915144456+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:36:12.915245214+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:36:49.902990347+01:00 [ERROR] GPU hung
2025-03-05T23:42:53.427343141+01:00 [INFO] ========== starting zed ==========
2025-03-05T23:42:53.835902546+01:00 [INFO] Presentation support: "VK_KHR_wayland_surface"
2025-03-05T23:42:53.836449277+01:00 [INFO] Presentation support: "VK_KHR_xcb_surface"
2025-03-05T23:42:53.836495316+01:00 [INFO] Presentation support: "VK_KHR_xlib_surface"
2025-03-05T23:42:53.83651905+01:00 [INFO] Enabling Vulkan Portability
2025-03-05T23:42:53.836539656+01:00 [INFO] Enabling color space support
2025-03-05T23:42:53.86909262+01:00 [INFO] Adapter: "AMD Radeon Graphics (RADV GFX1103_R1)"
2025-03-05T23:42:53.869365316+01:00 [INFO] Ray tracing is supported
2025-03-05T23:42:53.893433974+01:00 [INFO] perform;
2025-03-05T23:42:53.893489057+01:00 [INFO] authenticate;
2025-03-05T23:42:53.893557745+01:00 [INFO] read_command;
2025-03-05T23:42:53.894471122+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:53.897127945+01:00 [INFO] socket reader;
2025-03-05T23:42:53.897199388+01:00 [INFO] read_socket;
2025-03-05T23:42:53.897304104+01:00 [INFO] read_socket;
2025-03-05T23:42:53.897513565+01:00 [INFO] new;
2025-03-05T23:42:53.897811421+01:00 [INFO] read_socket;
2025-03-05T23:42:53.89792302+01:00 [INFO] read_socket;
2025-03-05T23:42:53.898007868+01:00 [INFO] read_socket;
2025-03-05T23:42:53.898266231+01:00 [INFO] read_socket;
2025-03-05T23:42:53.898325371+01:00 [INFO] keep_updated;
2025-03-05T23:42:53.901485104+01:00 [INFO] read_socket;
2025-03-05T23:42:53.902317088+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903250091+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903355518+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903455554+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903628838+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903721662+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903806931+01:00 [INFO] read_socket;
2025-03-05T23:42:53.903891539+01:00 [INFO] read_socket;
2025-03-05T23:42:53.90924527+01:00 [INFO] Opening main db
2025-03-05T23:42:53.925697222+01:00 [INFO] Opening main db
2025-03-05T23:42:53.936846018+01:00 [INFO] Using git binary path: None
2025-03-05T23:42:53.973314772+01:00 [INFO] set environment variables from shell:/bin/bash, path:/home/david/.local/bin:/home/david/.cargo/bin:/home/david/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
2025-03-05T23:42:54.054660085+01:00 [INFO] No prompt template overrides directory found at /home/david/.local/share/zed/prompt_overrides. Using built-in prompts.
2025-03-05T23:42:54.087945302+01:00 [INFO] extensions updated. loading 1, reloading 0, unloading 0
2025-03-05T23:42:54.143062982+01:00 [INFO] activate is not implemented on Linux, ignoring the call
2025-03-05T23:42:54.16700768+01:00 [INFO] Opening main db
2025-03-05T23:42:54.167902822+01:00 [INFO] perform;
2025-03-05T23:42:54.167937237+01:00 [INFO] authenticate;
2025-03-05T23:42:54.167974947+01:00 [INFO] read_command;
2025-03-05T23:42:54.168011586+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:54.168293572+01:00 [INFO] socket reader;
2025-03-05T23:42:54.168328117+01:00 [INFO] read_socket;
2025-03-05T23:42:54.168374323+01:00 [INFO] read_socket;
2025-03-05T23:42:54.173225034+01:00 [INFO] read_socket;
2025-03-05T23:42:54.174091703+01:00 [INFO] read_socket;
2025-03-05T23:42:54.174745595+01:00 [INFO] read_socket;
2025-03-05T23:42:54.175209021+01:00 [INFO] read_socket;
2025-03-05T23:42:54.191565424+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2025-03-05T23:42:54.194466193+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:54.195045054+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-05T23:42:54.19507504+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2025-03-05T23:42:54.195099796+01:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2025-03-05T23:42:54.250280865+01:00 [INFO] Using GPU: GpuSpecs { is_software_emulated: false, device_name: "AMD Radeon Graphics (RADV GFX1103_R1)", driver_name: "radv", driver_info: "Mesa 24.0.3-1pop1~1711635559~22.04~7a9f319" }
2025-03-05T23:42:54.260544077+01:00 [INFO] Opening main db
2025-03-05T23:42:54.28308336+01:00 [INFO] open paths ["/home/david/.config/zed/settings.json"]
2025-03-05T23:42:54.286805052+01:00 [INFO] Opening main db
2025-03-05T23:42:54.368282532+01:00 [INFO] attempting to start language server "json-language-server", path: "/home/david/.config/zed/settings.json", id: 1
2025-03-05T23:42:54.368832299+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-05T23:42:54.401967616+01:00 [INFO] using project environment variables shell launched in "/home/david/.config/zed/settings.json". PATH="/home/david/.local/bin:/home/david/.local/bin:/home/david/.cargo/bin:/home/david/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-05T23:42:54.402928711+01:00 [INFO] fetching latest version of language server "json-language-server"
2025-03-05T23:42:54.403092027+01:00 [INFO] Node runtime install_if_needed
2025-03-05T23:42:54.417006328+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-05T23:42:54.418113266+01:00 [INFO] attempting to start language server "rust-analyzer", path: "/home/david/repos/string_interner", id: 2
2025-03-05T23:42:54.441737506+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-05T23:42:54.458575169+01:00 [INFO] using project environment variables shell launched in "/home/david/repos/string_interner". PATH="/home/david/.local/bin:/home/david/.local/bin:/home/david/.cargo/bin:/home/david/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
2025-03-05T23:42:54.458664937+01:00 [INFO] found rust-analyzer in PATH. trying to run `rust-analyzer --help`
2025-03-05T23:42:54.471140972+01:00 [ERROR] persisting editor selections for editor 4294967498, workspace WorkspaceId(4)

Caused by:
    Sqlite call failed with code 787 and message: Some("FOREIGN KEY constraint failed")
2025-03-05T23:42:54.600450202+01:00 [ERROR] failed to run rust-analyzer after detecting it in PATH: binary: "/home/david/.cargo/bin/rust-analyzer": exit status: 1, stdout: "", stderr: "error: Unknown binary 'rust-analyzer' in official toolchain 'stable-x86_64-unknown-linux-gnu'.\n"
2025-03-05T23:42:54.600542404+01:00 [INFO] fetching latest version of language server "rust-analyzer"
2025-03-05T23:42:54.610264495+01:00 [INFO] Node runtime install_if_needed
2025-03-05T23:42:54.672305828+01:00 [INFO] starting language server process. binary path: "/home/david/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/", args: ["/home/david/.local/share/zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2025-03-05T23:42:55.042835138+01:00 [INFO] Node runtime install_if_needed
2025-03-05T23:42:55.107873389+01:00 [INFO] Node runtime install_if_needed
2025-03-05T23:42:55.173560453+01:00 [INFO] language server "json-language-server" is already installed
2025-03-05T23:42:55.173673865+01:00 [INFO] starting language server process. binary path: "/home/david/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/home/david/.config/zed", args: ["/home/david/.local/share/zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2025-03-05T23:42:55.232470417+01:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2025-03-05T23:42:55.232598496+01:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2025-03-05T23:42:55.789548861+01:00 [INFO] downloading language server "rust-analyzer"
2025-03-05T23:42:55.790136048+01:00 [INFO] starting language server process. binary path: "/home/david/.local/share/zed/languages/rust-analyzer/rust-analyzer-2025-03-04", working directory: "/home/david/repos/string_interner", args: []
2025-03-05T23:42:57.035417145+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.035476727+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.036632626+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.036655589+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.05790909+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.057988579+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.059108972+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.059161149+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.073895403+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.073950075+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.074652117+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.074673296+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.08811757+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.088172893+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.089935085+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.089956425+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.103546872+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.103601364+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.104542482+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.104586333+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.120466979+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.120523084+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.121954127+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.121977761+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.136408498+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.136464302+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.137483465+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.137538308+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.138682636+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.138703825+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.154351936+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.154416256+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.155336134+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.155382691+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.170316137+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.170372132+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.171442942+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.171496502+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.18604025+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.186094962+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.18695599+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.187003649+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.202597218+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.202650878+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.203490918+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.203510855+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.218952901+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.21901138+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.219952428+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.219995258+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.23527489+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.235333229+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.236090834+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.236117224+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.252727682+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.252789938+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.25385637+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.253877249+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.268215463+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.268289231+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.26917739+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.269216413+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.284705476+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.28476107+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.285632859+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.285677733+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.300874771+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.300931627+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.301800761+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.30183235+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.317453239+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.317515335+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.318671786+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.318692885+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.323758207+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.323810294+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.334351866+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.33441301+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.351288634+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.35135083+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.35227656+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.352296256+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.358113754+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.358159579+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.367681075+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.367737541+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.373910452+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.373956979+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.383380542+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.383436456+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.38893082+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.388979481+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.399644603+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.39970178+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.404937781+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.404987123+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.416926135+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.416985406+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.42105598+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.421111684+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.432664414+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.432723905+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.436927337+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.436988071+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.448848907+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.44890415+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.452946902+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.452994912+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.46596528+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.466021756+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.468835241+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.468891877+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.484074108+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.484134951+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.485156349+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.48520507+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.498424434+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.498476692+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.500647767+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.500670961+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.515861887+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.515918032+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.516765165+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.516787386+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.531715211+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.531767599+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.532742581+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.532765063+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.551799819+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.551833933+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.552433002+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.552457037+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.558456664+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.558512328+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.56659652+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.566650501+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.572895777+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.57295616+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.581203576+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.581266093+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.588912957+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.588960486+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.647073581+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.647130487+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.654965994+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.65501717+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:57.665165286+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:42:57.665222633+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:42:59.273918181+01:00 [INFO] perform;
2025-03-05T23:42:59.273995425+01:00 [INFO] perform;
2025-03-05T23:42:59.274023988+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274047151+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274088509+01:00 [INFO] perform;
2025-03-05T23:42:59.27413215+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274171644+01:00 [INFO] read_command;
2025-03-05T23:42:59.274190309+01:00 [INFO] perform;
2025-03-05T23:42:59.274210587+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274238279+01:00 [INFO] read_command;
2025-03-05T23:42:59.274277622+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.274296237+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.274317617+01:00 [INFO] perform;
2025-03-05T23:42:59.274356349+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274401624+01:00 [INFO] read_command;
2025-03-05T23:42:59.274430468+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.274455655+01:00 [INFO] read_command;
2025-03-05T23:42:59.274503143+01:00 [INFO] read_command;
2025-03-05T23:42:59.274529032+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.27464072+01:00 [INFO] socket reader;
2025-03-05T23:42:59.274694982+01:00 [INFO] read_socket;
2025-03-05T23:42:59.274732011+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.27475816+01:00 [INFO] perform;
2025-03-05T23:42:59.274783467+01:00 [INFO] socket reader;
2025-03-05T23:42:59.274808133+01:00 [INFO] read_socket;
2025-03-05T23:42:59.274832659+01:00 [INFO] read_socket;
2025-03-05T23:42:59.274858327+01:00 [INFO] authenticate;
2025-03-05T23:42:59.274888593+01:00 [INFO] socket reader;
2025-03-05T23:42:59.274916946+01:00 [INFO] read_socket;
2025-03-05T23:42:59.274950379+01:00 [INFO] read_socket;
2025-03-05T23:42:59.274984963+01:00 [INFO] perform;
2025-03-05T23:42:59.275017384+01:00 [INFO] authenticate;
2025-03-05T23:42:59.275051127+01:00 [INFO] read_command;
2025-03-05T23:42:59.275077667+01:00 [INFO] read_socket;
2025-03-05T23:42:59.27510092+01:00 [INFO] read_command;
2025-03-05T23:42:59.275150412+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.275214913+01:00 [INFO] socket reader;
2025-03-05T23:42:59.275236914+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275273603+01:00 [INFO] socket reader;
2025-03-05T23:42:59.275300082+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275329116+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275370423+01:00 [INFO] send_secondary_commands;
2025-03-05T23:42:59.275399598+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275459099+01:00 [INFO] socket reader;
2025-03-05T23:42:59.275484236+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275522407+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275603679+01:00 [INFO] socket reader;
2025-03-05T23:42:59.275623166+01:00 [INFO] read_socket;
2025-03-05T23:42:59.275660986+01:00 [INFO] read_socket;
2025-03-05T23:42:59.27717264+01:00 [INFO] read_socket;
2025-03-05T23:42:59.278781766+01:00 [INFO] read_socket;
2025-03-05T23:42:59.280357579+01:00 [INFO] read_socket;
2025-03-05T23:42:59.281897846+01:00 [INFO] read_socket;
2025-03-05T23:42:59.283542328+01:00 [INFO] read_socket;
2025-03-05T23:42:59.285784417+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287396007+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287443997+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287491495+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287513657+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287561757+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287607532+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287656534+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287694795+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287740881+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287780726+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287835899+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287879921+01:00 [INFO] Initializing default prettier with plugins {}
2025-03-05T23:42:59.287905188+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287934733+01:00 [INFO] read_socket;
2025-03-05T23:42:59.287971752+01:00 [INFO] new;
2025-03-05T23:42:59.28801857+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288059647+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288155325+01:00 [INFO] read_socket;
2025-03-05T23:42:59.28819546+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288223433+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288289636+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288338948+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288367251+01:00 [INFO] new;
2025-03-05T23:42:59.28839869+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288523643+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288561444+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288627167+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288683572+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288709671+01:00 [INFO] read_socket;
2025-03-05T23:42:59.288739787+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289062961+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289117052+01:00 [INFO] keep_updated;
2025-03-05T23:42:59.289160994+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289181082+01:00 [INFO] new;
2025-03-05T23:42:59.289233149+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289272993+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289292069+01:00 [INFO] keep_updated;
2025-03-05T23:42:59.28934625+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289366498+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289397947+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289428544+01:00 [INFO] read_socket;
2025-03-05T23:42:59.28945349+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289505357+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289545262+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289569537+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289603491+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289647092+01:00 [INFO] keep_updated;
2025-03-05T23:42:59.289823231+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289883864+01:00 [INFO] read_socket;
2025-03-05T23:42:59.289928247+01:00 [INFO] read_socket;
2025-03-05T23:43:01.50460073+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.504681041+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.518697433+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.518773555+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.519787389+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.519855506+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.526834153+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.52692899+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.537909902+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.53800493+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.540693652+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.540784352+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.548807829+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.548891025+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.558483524+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.558582959+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.5693757+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.569468804+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.572705801+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.572767266+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.580914946+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.58099769+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.588896956+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.588980402+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.601271962+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.601386687+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.611655048+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.61174674+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.612894373+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.612937554+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.620790844+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.620875903+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.636075416+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.636167538+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.638738951+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.638771542+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.646910195+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.646953356+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.655705575+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.655753043+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.667530714+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.66759814+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.678793804+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.678876017+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.679879692+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.679908546+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.686849151+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.686893123+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.699730914+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.699797248+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.710269029+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.71031763+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.71119538+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.711215016+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.718908749+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.718952691+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.7331706+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.733220362+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.734879983+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.73492144+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.7489366+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.748986303+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.750801985+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.750834336+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.766076458+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.766124588+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.767568044+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.767600485+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.782816749+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.78286585+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.783680442+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.783724274+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.799166871+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.799216493+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.799934705+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.799953751+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.816160886+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.81621126+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.816981459+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.817019089+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.832331452+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.832385934+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.833134312+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.833172293+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.849640145+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.849689467+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.930168642+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.930294307+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:01.939058859+01:00 [INFO] Using surface present mode MAILBOX
2025-03-05T23:43:01.939146433+01:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-03-05T23:43:08.134771664+01:00 [INFO] perform;
2025-03-05T23:43:08.134829232+01:00 [INFO] authenticate;
2025-03-05T23:43:08.134867263+01:00 [INFO] perform;
2025-03-05T23:43:08.134948785+01:00 [INFO] perform;
2025-03-05T23:43:08.134998798+01:00 [INFO] authenticate;
2025-03-05T23:43:08.135037841+01:00 [INFO] read_command;
2025-03-05T23:43:08.135084398+01:00 [INFO] send_secondary_commands;
2025-03-05T23:43:08.135113102+01:00 [INFO] perform;
2025-03-05T23:43:08.135204813+01:00 [INFO] authenticate;
2025-03-05T23:43:08.135330538+01:00 [INFO] read_command;
2025-03-05T23:43:08.135383317+01:00 [INFO] authenticate;
2025-03-05T23:43:08.135507198+01:00 [INFO] read_command;
2025-03-05T23:43:08.13559356+01:00 [INFO] send_secondary_commands;
2025-03-05T23:43:08.135661016+01:00 [INFO] read_command;
2025-03-05T23:43:08.135759389+01:00 [INFO] send_secondary_commands;
2025-03-05T23:43:08.13585081+01:00 [INFO] socket reader;
2025-03-05T23:43:08.13587734+01:00 [INFO] read_socket;
2025-03-05T23:43:08.135922564+01:00 [INFO] read_socket;
2025-03-05T23:43:08.13596335+01:00 [INFO] send_secondary_commands;
2025-03-05T23:43:08.135996542+01:00 [INFO] socket reader;
2025-03-05T23:43:08.13602742+01:00 [INFO] read_socket;
2025-03-05T23:43:08.136061945+01:00 [INFO] socket reader;
2025-03-05T23:43:08.136087653+01:00 [INFO] read_socket;
2025-03-05T23:43:08.136113691+01:00 [INFO] read_socket;
2025-03-05T23:43:08.136155539+01:00 [INFO] read_socket;
2025-03-05T23:43:08.1362095+01:00 [INFO] socket reader;
2025-03-05T23:43:08.136232834+01:00 [INFO] read_socket;
2025-03-05T23:43:08.1362682+01:00 [INFO] read_socket;
2025-03-05T23:43:08.138333909+01:00 [INFO] read_socket;
2025-03-05T23:43:08.139861662+01:00 [INFO] read_socket;
2025-03-05T23:43:08.141374418+01:00 [INFO] read_socket;
2025-03-05T23:43:08.142858761+01:00 [INFO] read_socket;
2025-03-05T23:43:08.142919775+01:00 [INFO] read_socket;
2025-03-05T23:43:08.142960291+01:00 [INFO] read_socket;
2025-03-05T23:43:08.142987492+01:00 [INFO] read_socket;
2025-03-05T23:43:08.143181083+01:00 [INFO] read_socket;
2025-03-05T23:43:08.14321137+01:00 [INFO] read_socket;
2025-03-05T23:43:08.14330223+01:00 [INFO] read_socket;
2025-03-05T23:43:08.143445076+01:00 [INFO] read_socket;
2025-03-05T23:43:08.143599345+01:00 [INFO] read_socket;
2025-03-05T23:43:08.143622077+01:00 [INFO] read_socket;
2025-03-05T23:43:08.144238108+01:00 [INFO] read_socket;
2025-03-05T23:43:08.144408005+01:00 [INFO] read_socket;
2025-03-05T23:43:08.144525334+01:00 [INFO] read_socket;
2025-03-05T23:43:20.4076265+01:00 [ERROR] GPU hung
david@pop-os:~$ 
```
</details>

Note the: 2025-03-05T23:43:20.4076265+01:00 [ERROR] GPU hung

> Similar issue. OS freezes, screen goes black, and sometimes returns but is freezed. AMD system
> 
> ```
> zed --version
> Zed 0.176.2 efe4418b9a2f9279604b6e03a42f204363b0d16e  – /home/david/.local/zed.app/libexec/zed-editor
> 
> 
>                              OS: Pop 22.04 jammy
>                              Kernel: x86_64 Linux 6.9.3-76060903-generic
>          #####               Uptime: 6h 13m
>         #######              Packages: Unknown
>         ##O#O##              Shell: bash 5.1.16
>         #######              Resolution: 3840x2160
>       ###########            DE: GNOME 41.7
>      #############           WM: Mutter
>     ###############          WM Theme: Pop
>     ################         GTK Theme: Pop-dark [GTK2/3]
>    #################         Icon Theme: Pop
>  #####################       Font: Fira Sans Semi-Light 10
>  #####################       Disk: 218G / 3.7T (7%)
>    #################         CPU: AMD Ryzen 9 8945HS w/ Radeon 780M Graphics @ 16x 5.263GHz
>                              GPU: AMD Radeon Graphics (radeonsi, gfx1103_r1, LLVM 15.0.7, DRM 3.57, 6.9.3-76060903-generic)
>                              RAM: 10774MiB / 94320MiB
> ```
> 
> Logs
> 
> Note the: 2025-03-05T23:43:20.4076265+01:00 [ERROR] GPU hung

Pretty much the same, AMD cpu + AMD gpu
`Zed 0.176.3 fbfa89356e8e74041585f721665f3016b0ba498a`

Hey all, I have the same issue. 

My Zed version: `Zed 0.177.9 4da6dfa89a770f081213f17463ac121f46a6a825 `


My last few lines in the `Zed.log`:
```
2025-03-16T13:55:01.218669783+11:00 [INFO] Initializing default prettier with plugins {}
2025-03-16T13:55:01.220371118+11:00 [INFO] read_socket;
2025-03-16T13:55:01.222767599+11:00 [INFO] read_socket;
2025-03-16T13:55:01.224969144+11:00 [INFO] read_socket;
2025-03-16T13:55:01.226851379+11:00 [INFO] read_socket;
2025-03-16T13:55:01.2286091+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228638585+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228674984+11:00 [INFO] read_socket;
2025-03-16T13:55:01.22869966+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228728915+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228755174+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228841807+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228883315+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228945912+11:00 [INFO] read_socket;
2025-03-16T13:55:01.228974556+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229022757+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229128075+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229169011+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229423429+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229495515+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229550047+11:00 [INFO] read_socket;
2025-03-16T13:55:01.229598769+11:00 [INFO] read_socket;
2025-03-16T13:55:01.22994047+11:00 [INFO] read_socket;
2025-03-16T13:55:01.230119517+11:00 [INFO] read_socket;
2025-03-16T13:55:18.476823341+11:00 [ERROR] GPU hung

```

--- a/crates/gpui/src/platform/blade/blade_renderer.rs
+++ b/crates/gpui/src/platform/blade/blade_renderer.rs
@@ -18,7 +18,7 @@ use std::{mem, sync::Arc};
 const MAX_FRAME_TIME_MS: u32 = 10000;
 // Use 4x MSAA, all devices support it.
 // https://developer.apple.com/documentation/metal/mtldevice/1433355-supportstexturesamplecount
-const PATH_SAMPLE_COUNT: u32 = 4;
+const DEFAULT_PATH_SAMPLE_COUNT: u32 = 4;
 
 #[repr(C)]
 #[derive(Clone, Copy, Pod, Zeroable)]
@@ -130,7 +130,7 @@ struct BladePipelines {
 }
 
 impl BladePipelines {
-    fn new(gpu: &gpu::Context, surface_info: gpu::SurfaceInfo) -> Self {
+    fn new(gpu: &gpu::Context, surface_info: gpu::SurfaceInfo, path_sample_count: u32) -> Self {
         use gpu::ShaderData as _;
 
         log::info!(
@@ -211,7 +211,7 @@ impl BladePipelines {
                     write_mask: gpu::ColorWrites::default(),
                 }],
                 multisample_state: gpu::MultisampleState {
-                    sample_count: PATH_SAMPLE_COUNT,
+                    sample_count: path_sample_count,
                     ..Default::default()
                 },
             }),
@@ -322,6 +322,7 @@ pub struct BladeRenderer {
     atlas_sampler: gpu::Sampler,
     #[cfg(target_os = "macos")]
     core_video_texture_cache: CVMetalTextureCache,
+    path_sample_count: u32,
 }
 
 impl BladeRenderer {
@@ -347,13 +348,18 @@ impl BladeRenderer {
             name: "main",
             buffer_count: 2,
         });
-        let pipelines = BladePipelines::new(&context.gpu, surface.info());
+        // workaround for https://github.com/zed-industries/zed/issues/26143
+        let path_sample_count = std::env::var("ZED_PATH_SAMPLE_COUNT")
+            .ok()
+            .and_then(|v| v.parse().ok())
+            .unwrap_or(DEFAULT_PATH_SAMPLE_COUNT);
+        let pipelines = BladePipelines::new(&context.gpu, surface.info(), path_sample_count);
         let instance_belt = BufferBelt::new(BufferBeltDescriptor {
             memory: gpu::Memory::Shared,
             min_chunk_size: 0x1000,
             alignment: 0x40, // Vulkan `minStorageBufferOffsetAlignment` on Intel Xe
         });
-        let atlas = Arc::new(BladeAtlas::new(&context.gpu, PATH_SAMPLE_COUNT));
+        let atlas = Arc::new(BladeAtlas::new(&context.gpu, path_sample_count));
         let atlas_sampler = context.gpu.create_sampler(gpu::SamplerDesc {
             name: "atlas",
             mag_filter: gpu::FilterMode::Linear,
@@ -382,6 +388,7 @@ impl BladeRenderer {
             atlas_sampler,
             #[cfg(target_os = "macos")]
             core_video_texture_cache,
+            path_sample_count,
         })
     }
 
@@ -389,6 +396,15 @@ impl BladeRenderer {
         if let Some(last_sp) = self.last_sync_point.take() {
             if !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS) {
                 log::error!("GPU hung");
+                #[cfg(target_os = "linux")]
+                if self.gpu.device_information().driver_name == "radv" {
+                    log::error!("there's a known bug with amdgpu/radv, try setting ZED_PATH_SAMPLE_COUNT=0 as a workaround");
+                    log::error!("if that helps you're running into https://github.com/zed-industries/zed/issues/26143");
+                }
+                log::error!(
+                    "your device information is: {:?}",
+                    self.gpu.device_information()
+                );
                 while !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS) {}
             }
         }
@@ -428,7 +444,8 @@ impl BladeRenderer {
             self.gpu
                 .reconfigure_surface(&mut self.surface, self.surface_config);
             self.pipelines.destroy(&self.gpu);
-            self.pipelines = BladePipelines::new(&self.gpu, self.surface.info());
+            self.pipelines =
+                BladePipelines::new(&self.gpu, self.surface.info(), self.path_sample_count);
         }
     }
 

--- a/docs/src/linux.md
+++ b/docs/src/linux.md
@@ -111,6 +111,7 @@ On some systems the file `/etc/prime-discrete` can be used to enforce the use of
 
 On others, you may be able to the environment variable `DRI_PRIME=1` when running Zed to force the use of the discrete GPU.
 
+If you're using an AMD GPU and Zed crashes when selecting long lines, try setting the `ZED_PATH_SAMPLE_COUNT=0` environment variable. (See [#26143](https://github.com/zed-industries/zed/issues/26143))
 If you're using an AMD GPU, you might get a 'Broken Pipe' error. Try using the RADV or Mesa drivers. (See [#13880](https://github.com/zed-industries/zed/issues/13880))
 
 If you are using Mesa, and want more control over which GPU is selected you can run `MESA_VK_DEVICE_SELECT=list zed --foreground` to get a list of available GPUs and then export `MESA_VK_DEVICE_SELECT=xxxx:yyyy` to choose a specific device.

